go-controller/pkg/ovn/ovn.go

```diff
@@ -553,6 +574,10 @@ func (oc *Controller) WatchNodes() error {
 		},
 		DeleteFunc: func(obj interface{}) {
 			node := obj.(*kapi.Node)
+
+			if noHostSubnet := noHostSubnet(node); noHostSubnet {
```

Comment:
@JacobTanenbaum I think we should remove this chunk, otherwise we'll never clear deleted node entries from the logical switch cache. It's fine to run the cleanup even for nodes that shouldn't have hostsubnets.

Reply:
fixed